src/benchmark/tpcds_benchmark.cpp

```diff
@@ -0,0 +1,127 @@
+#include "scheduler/topology.hpp"
+#include "sql/sql_pipeline.hpp"
+#include "sql/sql_pipeline_builder.hpp"
+#include "storage/chunk_encoder.hpp"
+#include "storage/storage_manager.hpp"
+#include "utils/assert.hpp"
+#include "utils/sqlite_wrapper.hpp"
+#include "utils/timer.hpp"
+#include "visualization/lqp_visualizer.hpp"
+#include "visualization/pqp_visualizer.hpp"
+
+using namespace opossum;  // NOLINT
+
+int main(int argc, char* argv[]) {
+  auto cli_options = opossum::BenchmarkRunner::get_basic_cli_options("TPCDS Benchmark");
+
+  // clang-format off
+  cli_options.add_options()
+    ("s,scale", "Database scale factor (1 ~ 1GB)", cxxopts::value<int>()->default_value("1"));
+  // clang-format on
+
+  std::shared_ptr<opossum::BenchmarkConfig> config;
+  int scale_factor;
+
+  if (opossum::CLIConfigParser::cli_has_json_config(argc, argv)) {
+    // JSON config file was passed in
+    const auto json_config = opossum::CLIConfigParser::parse_json_config_file(argv[1]);
+    scale_factor = json_config.value("scale", 1);
+    config = std::make_shared<opossum::BenchmarkConfig>(
+        opossum::CLIConfigParser::parse_basic_options_json_config(json_config));
+
+  } else {
+    // Parse regular command line args
+    const auto cli_parse_result = cli_options.parse(argc, argv);
+
+    if (CLIConfigParser::print_help_if_requested(cli_options, cli_parse_result)) {
+      return 0;
+    }
+    scale_factor = cli_parse_result["scale"].as<int>();
+
+    config =
+        std::make_shared<opossum::BenchmarkConfig>(opossum::CLIConfigParser::parse_basic_cli_options(cli_parse_result));
+  }
+
+  Assert(scale_factor == 1, "For now, TPC-DS benchmark only supports scale factor 1");
+  Assert(!config->enable_scheduler, "Multithreaded benchmark execution is currently not supported for TPC-DS");
+
+  auto context = opossum::BenchmarkRunner::create_context(*config);
+
+  std::cout << "- TPCDS scale factor is " << scale_factor << std::endl;
+
+  // TPC-DS FileBasedQueryGenerator specification
+  std::optional<std::unordered_set<std::string>> query_subset;
+  const auto query_filename_blacklist = std::unordered_set<std::string>{};
+  std::string query_path = "resources/benchmark/tpcds/queries";
+  std::string table_path = "resources/benchmark/tpcds/tables";
+
+  Assert(std::filesystem::is_directory(query_path), "Query path (" + query_path + ") has to be a directory.");
+  Assert(std::filesystem::is_directory(table_path), "Table path (" + table_path + ") has to be a directory.");
+
+  auto query_generator =
+      std::make_unique<FileBasedQueryGenerator>(*config, query_path, query_filename_blacklist, query_subset);
+  auto table_generator = std::make_unique<FileBasedTableGenerator>(config, table_path);
+  auto benchmark_runner = BenchmarkRunner{*config, std::move(query_generator), std::move(table_generator), context};
+
+  if (config->verify) {
+    // TODO(anyone) encapsulate this code somewhere else since it us used for the TPC-DS and JOIN ORDER BENCHMARK
```

Comment:
How much effort do you expect this to be?

Reply:
It's done now.